### PR TITLE
[GH-2384] Update resource names

### LINKED ISSUE #2384: MahApps.Metro consistent style and template names/keys
MahApps.Metro styles and template keys are non consostent.
### Expected outcome

Consistent names/keys for all styles and templates.

e.g.

`Win10WindowButtonCommandsTemplate` -> `MahApps.Metro.Templates.WindowButtonCommands.Win10`
`MetroFlatButton` -> `MahApps.Metro.Styles.FlatButton`

etc

### PR DESCRIPTION
**Pull request by Evangelink**
**Describe the changes you have made to improve this project**

Update various resource names.

/!\ This PR will need a big in-depth review /!\ 

I have tried to create relatively small commits (except for style) per resource type and I have tried to follow as closely as possible the pattern we discussed about.

**Final naming convention is:**

`MahApp.`
\+ the resource type which is one of `Colors.`, `Brushes.`, `Templates.`, `Sizes.`, `Fonts.`, `Storyboard.`, `Styles.`
\+ the target control for `Templates` and `Styles` (e.g. `Slider.`, `Window.`...)
\+ an optional specifier which gives more context (e.g. `Slider` for `MahApps.Styles.Thumb` which helps understanding that this thumb style is for the context of a slider)
\+ an optional category (e.g. `Metro`, `Flat`, `Square`, `Win10`...)

**Closed Issues**

Fix #2384

### CODE REVIEW

File: src/MahApps.Metro/Styles/Controls.StatusBar.xaml [hunk 1]
```diff
@@ -29,7 +29,7 @@
         <Setter Property="VerticalContentAlignment" Value="Center" />
     </Style>
 
-    <Style x:Key="MetroStatusBarSeparator" TargetType="{x:Type Separator}">
+    <Style x:Key="MahApps.Styles.Separator.MahApps.Styles.StatusBar.Metro" TargetType="{x:Type Separator}">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.StatusBar.xaml]: Shouldn't be this like that? -> `MahApps.Styles.StatusBar.Separator.Metro`

File: src/MahApps.Metro/Styles/Controls.ComboBox.xaml [hunk 2]
```diff
@@ -198,7 +198,7 @@
         </Setter>
     </Style>
 
-    <Style x:Key="MetroComboBoxDropDownToggleButtonStyle" TargetType="{x:Type ToggleButton}">
+    <Style x:Key="MahApps.Styles.ToggleButton.MahApps.Styles.ComboBox.MetroDropDown" TargetType="{x:Type ToggleButton}">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.ComboBox.xaml]: Shouldn't be this like that? -> `MahApps.Styles.ComboBox.ToggleButton.Metro`

File: src/MahApps.Metro/Styles/Controls.Expander.xaml [hunk 3]
```diff
@@ -303,7 +303,7 @@
         </Setter>
     </Style>
 
-    <Storyboard x:Key="MetroExpanderExpandStoryboard">
+    <Storyboard x:Key="MahApps.Storyboard.MahApps.Styles.Expander.MetroExpand">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.Expander.xaml]: Shouldn't be this like that? -> `MahApps.Storyboard.Expander.MetroExpand`

File: src/MahApps.Metro/Styles/Controls.Expander.xaml [hunk 4]
```diff
@@ -313,7 +313,7 @@
         </ObjectAnimationUsingKeyFrames>
     </Storyboard>
 
-    <Storyboard x:Key="MetroExpanderCollapseStoryboard">
+    <Storyboard x:Key="MahApps.Storyboard.MahApps.Styles.Expander.MetroCollapse">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.Expander.xaml]: Shouldn't be this like that? -> `MahApps.Storyboard.Expander.MetroCollapse`

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 5]
```diff
@@ -4,19 +4,19 @@
                     xmlns:System="clr-namespace:System;assembly=mscorlib"
                     xmlns:po="http://schemas.microsoft.com/winfx/2006/xaml/presentation/options">
 
-    <Storyboard x:Key="MetroScrollBarThumbMouseOverStoryboard" po:Freeze="True">
+    <Storyboard x:Key="MahApps.Storyboard.MahApps.Styles.ScrollBar.MetroThumbMouseOver" po:Freeze="True">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: Shouldn't be this like that? -> `MahApps.Storyboard.ScrollBar.MetroThumbMouseOver`

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 6]
```diff
@@ -4,19 +4,19 @@
...
         <DoubleAnimationUsingKeyFrames Storyboard.TargetName="MouseOverRectangle" Storyboard.TargetProperty="(UIElement.Opacity)">
             <EasingDoubleKeyFrame KeyTime="0:0:0.15" Value="0.6" />
         </DoubleAnimationUsingKeyFrames>
     </Storyboard>
 
-    <Storyboard x:Key="MetroScrollBarThumbPressedStoryboard" po:Freeze="True">
+    <Storyboard x:Key="MahApps.Storyboard.MahApps.Styles.ScrollBar.MetroThumbPressed" po:Freeze="True">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: Shouldn't be this like that? -> `MahApps.Storyboard.ScrollBar.MetroThumbPressed`

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 7]
```diff
@@ -4,19 +4,19 @@
...
         <DoubleAnimationUsingKeyFrames Storyboard.TargetName="PressedRectangle" Storyboard.TargetProperty="(UIElement.Opacity)">
             <EasingDoubleKeyFrame KeyTime="0:0:0.15" Value="0.8" />
         </DoubleAnimationUsingKeyFrames>
     </Storyboard>
 
-    <Style x:Key="MetroScrollBarRepeatButtonSmallStyle" TargetType="{x:Type RepeatButton}">
+    <Style x:Key="MahApps.Styles.RepeatButton.MahApps.Styles.ScrollBar.MetroSmall" TargetType="{x:Type RepeatButton}">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: Shouldn't be this like that? -> `MahApps.ScrollBar.RepeatButton.MetroSmall`

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 8]
```diff
@@ -64,7 +64,7 @@
         <Setter Property="Width" Value="{DynamicResource MahApps.Templates.ScrollBar.HorizontalRepeatButtonWidth}" />
     </Style>
 
-    <Style x:Key="MetroScrollBarRepeatButtonLargeStyle" TargetType="{x:Type RepeatButton}">
+    <Style x:Key="MahApps.Styles.RepeatButton.MahApps.Styles.ScrollBar.MetroLarge" TargetType="{x:Type RepeatButton}">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: Shouldn't be this like that? -> `MahApps.ScrollBar.RepeatButton.MetroLarge`

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 9]
```diff
@@ -81,7 +81,7 @@
         </Setter>
     </Style>
 
-    <Style x:Key="MetroScrollBarThumbStyle" TargetType="{x:Type Thumb}">
+    <Style x:Key="MahApps.Styles.Thumb.MahApps.Styles.ScrollBar.Metro" TargetType="{x:Type Thumb}">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: Shouldn't be this like that? -> `MahApps.Styles.ScrollBar.Thumb.Metro`

File: src/MahApps.Metro/Styles/VS/Expander.xaml [hunk 10]
```diff
@@ -303,7 +303,7 @@
         </Setter>
     </Style>
 
-    <Storyboard x:Key="MetroExpanderExpandStoryboard">
+    <Storyboard x:Key="MahApps.Storyboard.MahApps.Styles.Expander.MetroExpand">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/VS/Expander.xaml]: Shouldn't be this like that? -> `MahApps.Storyboard.Expander.Thumb.MetroExpand`

File: src/MahApps.Metro/Styles/VS/Expander.xaml [hunk 11]
```diff
@@ -313,7 +313,7 @@
         </ObjectAnimationUsingKeyFrames>
     </Storyboard>
 
-    <Storyboard x:Key="MetroExpanderCollapseStoryboard">
+    <Storyboard x:Key="MahApps.Storyboard.MahApps.Styles.Expander.MetroCollapse">
```
Comment (punker76) [inline src/MahApps.Metro/Styles/VS/Expander.xaml]: Shouldn't be this like that? -> `MahApps.Storyboard.Expander.Thumb.MetroCollapse`

**Comment by Evangelink** [comment]
@punker76 I don't know, how would you name `MahApps.Sizes.ScrollBar.HorizontalRepeatButtonWidth`?

**Comment by Evangelink** [comment]
And also do you want the names to be `MahApps.Styles.Thumb.ScrollBar` for the style of the `Thumb` that belongs to the scrollbar or `MahApps.Styles.ScrollBar.Thumb`? For now I have always put the target type first.

**Comment by punker76** [comment]
> @punker76 I don't know, how would you name `MahApps.Sizes.ScrollBar.HorizontalRepeatButtonWidth`?

I must agree with you there.

**Comment by punker76** [comment]
> And also do you want the names to be `MahApps.Styles.Thumb.ScrollBar` for the style of the `Thumb` that belongs to the scrollbar or `MahApps.Styles.ScrollBar.Thumb`? For now I have always put the target type first.

Yes, you're right. But we must explain this explicitly with some samples for better understanding. Can you add such a description to this PR above? Thx!

**Comment by Evangelink** [comment]
I'll update the PR first post and add a similar comment at the end of the issue with the final naming:

`MahApp.`
+ the resource type which is one of `Colors.`, `Brushes.`, `Templates.`, `Sizes.`, `Fonts.`, `Storyboard.`, `Styles.`
+ the target control for `Templates` and `Styles` (e.g. `Slider.`, `Window.`...)
+ an optional specifier which gives more context (e.g. `Slider` for `MahApps.Styles.Thumb` which helps understanding that this thumb style is for the context of a slider)
+ an optional category (e.g. `Metro`, `Flat`, `Square`, `Win10`...)

But while writing this summary I have realized that not all my renames are actually matching. For example, shall I go for `MahApps.Styles.ButtonBase.MetroCircle` (current) or `MahApps.Styles.ButtonBase.Circle.Metro`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ProgressBars.xaml [hunk 12]
```diff
@@ -12,7 +12,7 @@
 
     <UserControl.Resources>
         <ResourceDictionary>
-            <Style BasedOn="{StaticResource MahApps.Metro.Styles.MetroHeader}" TargetType="{x:Type Controls:MetroHeader}">
+            <Style BasedOn="{StaticResource MahApps.Styles.MetroHeader.Metro}" TargetType="{x:Type Controls:MetroHeader}">
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ProgressBars.xaml]: Maybe some controls should be renamed as part of this breaking change version (probably not in this commit). It seems a bit weird to have `MahApps.Styles.MetroHeader.Metro`

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ProgressBars.xaml [hunk 13]
```diff
@@ -51,7 +51,7 @@
                             Minimum="0"
                             Orientation="Horizontal"
                             SmallChange="1"
-                            Style="{DynamicResource MahApps.Metro.Styles.Slider.Win10}"
+                            Style="{DynamicResource MahApps.Styles.Slider.Metro.Win10}"
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ProgressBars.xaml]: `Metro.Win10` -> `Win10`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ProgressBars.xaml [hunk 14]
```diff
@@ -131,7 +131,7 @@
...
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/ProgressBars.xaml]: `Metro.Win10` -> `Win10`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/SelectionExamples.xaml [hunk 15]
```diff
@@ -22,14 +22,14 @@
                              DisplayMemberPath="Name"
                              ItemsSource="{Binding Artists}"
                              SelectedIndex="0"
-                             Style="{StaticResource VirtualisedMetroListBox}" />
+                             Style="{StaticResource MahApps.Styles.ListBox.MetroVirtualized}" />
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/SelectionExamples.xaml]: Maybe this should be `MahApps.Styles.ListBox.Virtualized.Metro`, WDYT?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/SelectionExamples.xaml [hunk 16]
```diff
@@ -56,7 +56,7 @@
                               BorderThickness="1"
                               ItemsSource="{Binding Artists}"
                               SelectedIndex="0"
-                              Style="{StaticResource VirtualisedMetroListView}">
+                              Style="{StaticResource MahApps.Styles.ListView.MetroVirtualized}">
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/SelectionExamples.xaml]: Same here for `ListView` -> `MahApps.Styles.ListView.Virtualized.Metro`

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml [hunk 17]
```diff
@@ -110,7 +110,7 @@
                                     CommandParameter="{Binding RelativeSource={RelativeSource FindAncestor, AncestorType=TextBox}, Path=Text, Mode=OneWay}" />
                     </TextBox.InputBindings>
                     <TextBox.Style>
-                        <Style BasedOn="{StaticResource SearchMetroTextBox}" TargetType="{x:Type TextBox}">
+                        <Style BasedOn="{StaticResource MahApps.Styles.TextBox.MetroSearch}" TargetType="{x:Type TextBox}">
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml]: `MahApps.Styles.TextBox.MetroSearch` -> `MahApps.Styles.TextBox.Search.Metro`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml [hunk 18]
```diff
@@ -125,12 +125,12 @@
...
                          Controls:TextBoxHelper.ButtonCommandParameter="{Binding ElementName=test2, Path=Text}"
                          Controls:TextBoxHelper.Watermark="Enter parameter"
-                         Style="{DynamicResource SearchMetroTextBox}" />
+                         Style="{DynamicResource MahApps.Styles.TextBox.MetroSearch}" />
                 <TextBox Margin="{StaticResource ControlMargin}"
                          Controls:TextBoxHelper.ButtonCommand="{Binding TextBoxButtonCmd, Mode=OneWay}"
                          Controls:TextBoxHelper.ButtonContent="s"
                          Controls:TextBoxHelper.Watermark="Custom icon style"
-                         Style="{DynamicResource MetroButtonTextBox}" />
+                         Style="{DynamicResource MahApps.Styles.TextBox.MetroButton}" />
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml]: `MahApps.Styles.TextBox.MetroButton` -> `MahApps.Styles.TextBox.Button.Metro`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml [hunk 19]
```diff
@@ -187,7 +187,7 @@
                              SpellCheck.IsEnabled="True">
                     <!--
                         Workaround for: RichTextBox behaves oddly inside a ScrollViewer
-                        
+
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml]: Do you want me to undo this change?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml [hunk 20]
```diff
@@ -178,7 +178,7 @@
                              Controls:TextBoxHelper.ButtonContent="s"
                              IsDocumentEnabled="True"
                              SpellCheck.IsEnabled="True"
-                             Style="{StaticResource MahApps.Metro.Styles.ButtonRichTextBox}" />
+                             Style="{StaticResource MahApps.Styles.RichTextBox.MetroButton}" />
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml]: `MahApps.Styles.RichTextBox.MetroButton` -> `MahApps.Styles.RichTextBox.Button.Metro`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml [hunk 21]
```diff
@@ -244,13 +244,13 @@
...
                              Password="Win8"
-                             Style="{StaticResource Win8MetroPasswordBox}" />
+                             Style="{StaticResource MahApps.Styles.PasswordBox.Win8}" />
                 <PasswordBox Margin="{StaticResource ControlMargin}"
                              Controls:TextBoxHelper.ClearTextButton="{Binding RelativeSource={RelativeSource Self}, Path=(Controls:TextBoxHelper.HasText), Mode=OneWay}"
                              Controls:TextBoxHelper.IsWaitingForData="True"
                              Controls:TextBoxHelper.UseFloatingWatermark="True"
                              Controls:TextBoxHelper.Watermark="Revealed PasswordBox"
-                             Style="{StaticResource MetroButtonRevealedPasswordBox}" />
+                             Style="{StaticResource MahApps.Styles.PasswordBox.MetroButtonRevealed}" />
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml]: `MahApps.Styles.PasswordBox.MetroButtonRevealed` -> `MahApps.Styles.PasswordBox.ButtonRevealed.Metro`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml [hunk 22]
```diff
@@ -263,7 +263,7 @@
                              Controls:TextBoxHelper.ButtonContent="4"
                              Controls:TextBoxHelper.UseFloatingWatermark="True"
                              Controls:TextBoxHelper.Watermark="Custom icon style"
-                             Style="{DynamicResource MetroButtonPasswordBox}">
+                             Style="{DynamicResource MahApps.Styles.PasswordBox.MetroButton}">
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleViews/TextExamples.xaml]: `MahApps.Styles.PasswordBox.MetroButton` -> `MahApps.Styles.PasswordBox.Button.Metro`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleWindows/VSDemo.xaml [hunk 23]
```diff
@@ -28,7 +28,7 @@
 
     <Controls:MetroWindow.RightWindowCommands>
         <Controls:WindowCommands ShowSeparators="False">
-            <Button Style="{StaticResource VSWindowNotificationButtonStyle}" ToolTip="1 Notification">
+            <Button Style="{StaticResource MahApps.Styles.Button.VisualStudioWindowNotification}" ToolTip="1 Notification">
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleWindows/VSDemo.xaml]: `MahApps.Styles.Button.VisualStudioWindowNotification` -> `MahApps.Styles.Button.WindowNotification.VisualStudio`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleWindows/VSDemo.xaml [hunk 24]
```diff
@@ -38,7 +38,7 @@
                                                                              Width=12,
                                                                              Height=12}"
                      Controls:TextBoxHelper.Watermark="Quick Launch (Ctrl+Q)"
-                     Style="{DynamicResource VSWindowQuickLaunchTextBoxStyle}" />
+                     Style="{DynamicResource MahApps.Styles.TextBox.VisualStudioWindowQuickLaunch}" />
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/ExampleWindows/VSDemo.xaml]: `MahApps.Styles.TextBox.VisualStudioWindowQuickLaunch` -> `MahApps.Styles.TextBox.WindowQuickLaunch.VisualStudio`?

File: src/MahApps.Metro.Samples/MahApps.Metro.Demo/MainWindow.xaml [hunk 25]
```diff
@@ -320,7 +320,7 @@
 
         <StatusBar Grid.Row="2">
             <StatusBarItem>MahApps.Metro DEMO Application</StatusBarItem>
-            <Separator Style="{StaticResource MetroStatusBarSeparator}" />
+            <Separator Style="{StaticResource MahApps.Styles.Separator.MetroStatusBar}" />
```
Comment (Evangelink) [inline src/MahApps.Metro.Samples/MahApps.Metro.Demo/MainWindow.xaml]: `MahApps.Styles.Separator.MetroStatusBar` -> `MahApps.Styles.Separator.StatusBar.Metro`?

File: src/MahApps.Metro/Styles/Clean/CleanWindowButtonCommands.xaml [hunk 26]
```diff
@@ -6,9 +6,9 @@
         <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Clean/CleanWindowButtons.xaml" />
     </ResourceDictionary.MergedDictionaries>
 
-    <Style x:Key="CleanWindowButtonCommandsStyleKey" TargetType="{x:Type Controls:WindowButtonCommands}">
-        <Setter Property="DarkCloseButtonStyle" Value="{StaticResource DarkCleanWindowCloseButtonStyle}" />
-        <Setter Property="LightCloseButtonStyle" Value="{StaticResource LightCleanWindowCloseButtonStyle}" />
+    <Style x:Key="MahApps.Styles.WindowButtonCommands.Clean" TargetType="{x:Type Controls:WindowButtonCommands}">
+        <Setter Property="DarkCloseButtonStyle" Value="{StaticResource MahApps.Styles.Button.CleanWindowClose.Dark}" />
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Clean/CleanWindowButtonCommands.xaml]: `MahApps.Styles.Button.CleanWindowClose.Dark` -> `MahApps.Styles.Button.WindowClose.CleanDark? or `MahApps.Styles.Button.WindowClose.Dark.Clean` or `MahApps.Styles.Button.DarkWindowClose.Clean?`

File: src/MahApps.Metro/Styles/Clean/CleanWindowButtonCommands.xaml [hunk 27]
```diff
@@ -6,9 +6,9 @@
...
+        <Setter Property="LightCloseButtonStyle" Value="{StaticResource MahApps.Styles.Button.CleanWindowClose.Light}" />
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Clean/CleanWindowButtonCommands.xaml]: Same as previous

File: src/MahApps.Metro/Styles/Controls.DatePicker.xaml [hunk 28]
```diff
@@ -270,7 +270,7 @@
                                    HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
                                    VerticalAlignment="{TemplateBinding VerticalContentAlignment}"
                                    Foreground="{TemplateBinding Foreground}"
-                                   Style="{DynamicResource MahApps.Metro.Styles.MetroWatermarkTextBlock}"
+                                   Style="{DynamicResource MahApps.Styles.TextBlock.MetroWatermark}"
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Controls.DatePicker.xaml]: `MahApps.Styles.TextBlock.MetroWatermark` -> `MahApps.Styles.TextBlock.Watermark.Metro`?

File: src/MahApps.Metro/Styles/Controls.ListView.xaml [hunk 29]
```diff
@@ -2,7 +2,7 @@
                     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                     xmlns:Controls="clr-namespace:MahApps.Metro.Controls">
 
-    <Style x:Key="MetroGridViewScrollViewerStyle"
+    <Style x:Key="MahApps.Styles.ScrollViewer.MetroGridView"
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Controls.ListView.xaml]: `MahApps.Styles.ScrollViewer.MetroGridView` -> `MahApps.Styles.ScrollViewer.GridView.Metro`?

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 30]
```diff
@@ -4,22 +4,22 @@
...
 
-    <Storyboard x:Key="MetroScrollBarThumbPressedStoryboard" po:Freeze="True">
+    <Storyboard x:Key="MahApps.Storyboard.MetroScrollBarThumbPressed" po:Freeze="True">
         <DoubleAnimationUsingKeyFrames Storyboard.TargetName="PressedRectangle" Storyboard.TargetProperty="(UIElement.Opacity)">
             <EasingDoubleKeyFrame KeyTime="0:0:0.15" Value="0.8" />
         </DoubleAnimationUsingKeyFrames>
     </Storyboard>
 
-    <Style x:Key="MetroScrollBarRepeatButtonSmallStyle" TargetType="{x:Type RepeatButton}">
+    <Style x:Key="MahApps.Styles.RepeatButton.MetroScrollBarSmall" TargetType="{x:Type RepeatButton}">
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: `MahApps.Styles.RepeatButton.MetroScrollBarSmall` -> `MahApps.Styles.RepeatButton.ScrollBarSmall.Metro`?

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 31]
```diff
@@ -51,20 +51,20 @@
...
                 </ControlTemplate>
             </Setter.Value>
         </Setter>
         <Setter Property="VerticalAlignment" Value="Center" />
-        <Setter Property="Width" Value="{DynamicResource HorizontalScrollBarRepeatButtonWidth}" />
+        <Setter Property="Width" Value="{DynamicResource MahApps.Templates.ScrollBar.HorizontalRepeatButtonWidth}" />
     </Style>
 
-    <Style x:Key="MetroScrollBarRepeatButtonLargeStyle" TargetType="{x:Type RepeatButton}">
+    <Style x:Key="MahApps.Styles.RepeatButton.MetroScrollBarLarge" TargetType="{x:Type RepeatButton}">
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: `MahApps.Styles.RepeatButton.MetroScrollBarLarge` -> `MahApps.Styles.RepeatButton.ScrollBarLarge.Metro`?

File: src/MahApps.Metro/Styles/Controls.Scrollbars.xaml [hunk 32]
```diff
@@ -81,8 +81,8 @@
         </Setter>
     </Style>
 
-    <Style x:Key="MetroScrollBarThumbStyle" TargetType="{x:Type Thumb}">
-        <Setter Property="Background" Value="{DynamicResource ThumbBrush}" />
+    <Style x:Key="MahApps.Styles.Thumb.MetroScrollBar" TargetType="{x:Type Thumb}">
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Controls.Scrollbars.xaml]: `MahApps.Styles.Thumb.MetroScrollBar` -> `MahApps.Styles.Thumb.ScrollBar.Metro`?

File: src/MahApps.Metro/Styles/Sizes.xaml [hunk 33]
```diff
@@ -2,10 +2,10 @@
...
                     xmlns:System="clr-namespace:System;assembly=mscorlib">
 
-    <System:Double x:Key="MetroScrollBarHeight">14</System:Double>
-    <System:Double x:Key="MetroScrollBarWidth">14</System:Double>
+    <System:Double x:Key="MahApps.Sizes.ScrollBar.Height.Metro">14</System:Double>
+    <System:Double x:Key="MahApps.Sizes.ScrollBar.Width.Metro">14</System:Double>
 
-    <System:Double x:Key="HorizontalScrollBarRepeatButtonWidth">14</System:Double>
-    <System:Double x:Key="VerticalScrollBarRepeatButtonHeight">14</System:Double>
+    <System:Double x:Key="MahApps.Sizes.ScrollBar.HorizontalRepeatButtonWidth">14</System:Double>
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Sizes.xaml]: `MahApps.Sizes.ScrollBar.HorizontalRepeatButtonWidth` -> `MahApps.Sizes.ScrollBar.HorizontalRepeatButton.Width`?

File: src/MahApps.Metro/Styles/Sizes.xaml [hunk 34]
```diff
@@ -2,10 +2,10 @@
...
+    <System:Double x:Key="MahApps.Sizes.ScrollBar.VerticalRepeatButtonHeight">14</System:Double>
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/Sizes.xaml]: `MahApps.Sizes.ScrollBar.VerticalRepeatButtonHeight` -> `MahApps.Sizes.ScrollBar.VerticalRepeatButton.Height`?

File: src/MahApps.Metro/Styles/VS/Button.xaml [hunk 35]
```diff
@@ -1,10 +1,10 @@
 ﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
 
-    <Style x:Key="StandardButton" TargetType="Button">
-        <Setter Property="Background" Value="{DynamicResource BackgroundNormal}" />
-        <Setter Property="BorderBrush" Value="{DynamicResource BorderBrushNormal}" />
-        <Setter Property="Foreground" Value="{DynamicResource Foreground}" />
+    <Style x:Key="MahApps.Styles.Button.VisualStudioWindowStandard" TargetType="Button">
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/VS/Button.xaml]: `MahApps.Styles.Button.VisualStudioWindowStandard` -> `MahApps.Styles.Button.WindowStandard.VisualStudio`?

File: src/MahApps.Metro/Styles/VS/Button.xaml [hunk 36]
```diff
@@ -34,9 +34,9 @@
         </Setter>
     </Style>
 
-    <Style x:Key="LinkButton" TargetType="Button">
+    <Style x:Key="MahApps.Styles.Button.VisualStudioWindowLink" TargetType="Button">
```
Comment (Evangelink) [inline src/MahApps.Metro/Styles/VS/Button.xaml]: `MahApps.Styles.Button.VisualStudioWindowLink` -> `MahApps.Styles.Button.Window.Link.VisualStudio`?

**Comment by punker76** [comment]
@Evangelink I'm more and more in favor of removing Metro from the name. Because this Name already exists in the name of this library.

**Comment by Evangelink** [comment]
@punker76 Are you referring to the `.Metro` suffix? If so I can do a new commit.

Are you ok with the other changes I made?

I think we are close to the end with those changes :D

**Comment by punker76** [comment]
> @punker76 Are you referring to the `.Metro` suffix? If so I can do a new commit.

Yes.

> Are you ok with the other changes I made?
> 
> I think we are close to the end with those changes :D

Yes 😃 

Maybe these changes will be create some issues, but I think we are now clear with the naming style.

**Comment by Evangelink** [comment]
@punker76 Just a crazy thought, WDYT about creating a static class (or using resx) to host the name of the resources instead of relying on strings? This could be useful to provide some kind of strong typing.

**Comment by punker76** [comment]
@Evangelink You mean something similar to this? 

https://github.com/MahApps/MahApps.Metro/blob/a5d0c3afbbc78dc33fc3a120d270ad80a33c870a/src/MahApps.Metro/Controls/Helper/TextBoxHelper.cs#L19-L36

Mh... 🤔  has also Pros and Contras...

**Comment by Evangelink** [comment]
@punker76 yep (for what I meant and for it has pros and cons)

**Comment by mgnslndh** [comment]
I'm curious, what are the cons of strong typing the resource keys using the `ComponentResourceKey`?

**Comment by timunie** [comment]
@mgnslndh would this enable IntelliSense in XAML

Happy Coding
Tim
